### PR TITLE
Internationalization of Perseus Renderer

### LINKED ISSUE #90: Mock i18n functionality for text internal to the Perseus renderer
Currently any text internal to Perseus reaches out to the `i18n._` method, which has no access to any translation strings - also, we do not include the appropriate strings from the KA Crowdin to enable this.

### PR DESCRIPTION
**Pull request by rtibbles**
Fixes #90 

This PR depends on https://github.com/learningequality/kolibri/pull/1869, which includes some expanded building functionality to support things that are going on here.

This PR updates Perseus by two major versions, in order for the internal strings to be better aligned with those on the KA Crowdin repo.

As a result of this it was:
* Easier to make Perseus build from source
* Practical to clean up all the global imports and use the Webpack Provide Plugin in order to inject dependencies at build time

In addition, this PR features two utilities, one that should be run when we update the version of Perseus, which extracts strings from the Perseus source and creates a message object from them, the other downloads translations from KA Crowdin, and then uploads those translations to our crowdin so that we don't force our translators to duplicate the efforts KA has already made.

### CODE REVIEW

File: kolibri_exercise_perseus_plugin/generateKATranslations.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+ * search through them for interface strings that we need in the Perseus renderer,
+ * and then upload them to our Crowdin in order to leverage these translated strings
+ * in our own translations.
+ */
+
+
+/*
+ * Utility functions for generating the paths for locale files, both locally and on crowdin
+ */
+const localeFile = (lang, locale) => {
```
Comment (aronasorman) [inline kolibri_exercise_perseus_plugin/generateKATranslations.js:12]: What's an example of `lang` an `locale` here?

File: kolibri_exercise_perseus_plugin/generateKATranslations.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+  return lang + (locale ? '-' + locale.toUpperCase() : '');
+};
+
+// Path to the locale file for a language in the local locale folder
+const localLocaleFile = (lang, locale) => {
+  return './locale/' + localeFile(lang, locale);
+}
+
+// Which kolibri branch to upload translations to on Crowdin
+const kolibriReleaseBranch = 'release-v0.6.x';
```
Comment (aronasorman) [inline kolibri_exercise_perseus_plugin/generateKATranslations.js:35]: Does this map to the branch on CrowdIn? Do we need to edit this again when we have a new release branch?

File: kolibri_exercise_perseus_plugin/generateKATranslations.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+// Generate path for Kolibri crowdin translation upload
+const remoteLocaleFile = () => {
+  return `${kolibriReleaseBranch}/` + localeFile('en');
+}
+
+// Load upthe current English translations file, to give us basis for mapping
+// KA translations to Kolibri message ids
+const en = require(localLocaleFile('en'));
```
Comment (aronasorman) [inline kolibri_exercise_perseus_plugin/generateKATranslations.js:44]: What does require do in here? If I understand my JS correctly, we're loading a directory?

File: kolibri_exercise_perseus_plugin/generateKATranslations.js [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+const {
+  kaApiKey,
+  kolibriApiKey,
+} = require('../crowdinSecrets');
+const request = require('request-promise');
+const temp = require('temp').track();
+const ProgressBar = require('progress');
+
+// List of currently supported languages and associated language and locale codes.
+const languages = {
```
Comment (aronasorman) [inline kolibri_exercise_perseus_plugin/generateKATranslations.js:64]: Yeah, we should look into centralizing this into `le_utils` and making sure these are accessible on both python and JS.

File: kolibri_exercise_perseus_plugin/generateKATranslations.js [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+    const pofileDownloadPromise = new Promise((resolve, reject) => {
+      console.log(`Downloading KA Po files for ${language}`);
+      const request = https.get(zipUrl, response => {
+        // Write to a temp file
+        const zipFile = temp.openSync({ suffix: '.zip' }).path;
+        const zipStream = fs.createWriteStream(zipFile);
+        // Find the total size from the headers, so that we can provide a progress bar
+        const size = parseInt(response.headers['content-length'], 10);
+        // Make a nice progress bar!
+        const downloadBar = new ProgressBar('  downloading [:bar] :rate/bps :percent :etas', {
```
Comment (aronasorman) [inline kolibri_exercise_perseus_plugin/generateKATranslations.js:167]: Fancy 🕺

**Review by aronasorman — COMMENTED** [review]
Neutral review. Some questions regarding implementation, but the CrowdIn uploading looks good (and if it works it works!)

File: kolibri_exercise_perseus_plugin/generateKATranslations.js (hunk 1 above)
Comment (rtibbles) [reply kolibri_exercise_perseus_plugin/generateKATranslations.js:12]: lang `es`, locale `mx`.

File: kolibri_exercise_perseus_plugin/generateKATranslations.js (hunk 2 above)
Comment (rtibbles) [reply kolibri_exercise_perseus_plugin/generateKATranslations.js:35]: Yes, basically it maps to where this version of the exercise renderer is going to submit and receive its translations from.

File: kolibri_exercise_perseus_plugin/generateKATranslations.js (hunk 3 above)
Comment (rtibbles) [reply kolibri_exercise_perseus_plugin/generateKATranslations.js:44]: It loads the json file with the messages in.

File: kolibri_exercise_perseus_plugin/generateKATranslations.js (hunk 4 above)
Comment (rtibbles) [reply kolibri_exercise_perseus_plugin/generateKATranslations.js:64]: Indeed - also would be good to have a centralized list of 'available languages' for Kolibri.

**Review by DXCanas — APPROVED** [review]
Can't say I fully understand what's happening, but it seems like a lot of this is part of the perseus update. The rest seems to get language changes from perseus, parsing it into a way our setup will understand it, with the language being passed in by Kolibri's build process?

**Review by DXCanas — APPROVED** [review]
One day we'll find a better way

**Pull request merged**